### PR TITLE
Support MCRIBS derivatives

### LINKED ISSUE #840: [DCAN] Support MIRTK anatomical derivative structure
## Summary

Nibabies will use MIRTK instead of FreeSurfer when there is a T2w, but no T1w, so we will need to find certain files from the MIRTK folder structure in order to run the anatomical workflow. For FreeSurfer, we only need two sphere files, so hopefully MIRTK will be similar.

The two files we use from the FreeSurfer directory are `surf/lh.sphere.reg` and `surf/rh.sphere.reg`.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #840 and relates to #976.

## Changes proposed in this pull request

- Create interface, `xcp_d.interfaces.bids.CollectRegistrationFiles`, to collect FreeSurfer _or_ MCRIBs sphere and transform files for anatomical workflow.
- Remove `xcp_d.utils.bids.get_freesurfer_sphere`, as the functionality is now in `CollectRegistrationFiles`.
- Heavily refactor `xcp_d.utils.bids.get_freesurfer_dir` to support MCRIBS on top of FreeSurfer.

### CODE REVIEW

**Comment by tsalo** [comment]
@madisoth I think this should work. Would you be willing to try it out on some Nibabies+MCRIBS data?

**Review by madisoth — CHANGES_REQUESTED** [review]
```
class CollectRegistrationFiles(SimpleInterface):
    """Collect registration files for fsnative-to-fsLR transformation."""

    input_spec = _CollectRegistrationFilesInputSpec
    output_spec = _CollectRegistrationFilesOutputSpec

    def _run_interface(self, runtime):
        import os

        from pkg_resources import resource_filename as pkgrf
        from templateflow.api import get as get_template

        hemisphere = self.inputs.hemisphere
        hstr = f"{hemisphere.lower()}h"
        participant_id = self.inputs.participant_id
        if not participant_id.startswith("sub-"):
            participant_id = f"sub-{participant_id}"

        # Find the subject's sphere in the segmentation derivatives.
        # TODO: Collect from the preprocessing derivatives if they're a compliant version.
        # Namely, fMRIPrep >= 23.1.2, Nibabies >= 24.0.0a1.
        self._results["subject_sphere"] = os.path.join(
            self.inputs.segmentation_dir,
            participant_id,
            "surf",
            f"{hstr}.sphere.reg",
        )

        # NOTE: Why do we need the fsaverage mesh?
        # TODO: Replace with appropriate files.
        if self.inputs.software == "FreeSurfer":
            # Find the subject's sphere in the FreeSurfer derivatives.
            # TODO: Collect from the preprocessing derivatives if they're a compliant version.
            # Namely, fMRIPrep >= 23.1.2, Nibabies >= 24.0.0a1.
            self._results["subject_sphere"] = os.path.join(
                self.inputs.segmentation_dir,
                participant_id,
                "surf",
                f"{hstr}.sphere.reg",
            )
            # Load the fsaverage-164k sphere
            # FreeSurfer: tpl-fsaverage_hemi-?_den-164k_sphere.surf.gii
            self._results["source_sphere"] = str(
                get_template(
                    template="fsaverage",
                    space=None,
                    hemi=hemisphere,
                    density="164k",
                    desc=None,
                    suffix="sphere",
                )
            )
            # NOTE: Can we upload these to templateflow?
            # FreeSurfer: fs_?/fs_?-to-fs_LR_fsaverage.?_LR.spherical_std.164k_fs_?.surf.gii
            self._results["sphere_to_sphere"] = pkgrf(
                "xcp_d",
                (
                    f"data/standard_mesh_atlases/fs_{hemisphere}/"
                    f"fs_{hemisphere}-to-fs_LR_fsaverage.{hemisphere}_LR.spherical_std."
                    f"164k_fs_{hemisphere}.surf.gii"
                ),
            )
            # FreeSurfer: tpl-fsLR_hemi-?_den-32k_sphere.surf.gii
            self._results["target_sphere"] = str(
                get_template(
                    template="fsLR",
                    space=None,
                    hemi=hemisphere,
                    density="32k",
                    desc=None,
                    suffix="sphere",
                )
            )

        elif self.inputs.software == "MCRIBS":
            # Find the subject's sphere in the MCRIBS derivatives.
            # TODO: Collect from the preprocessing derivatives if they're a compliant version.
            # Namely, fMRIPrep >= 23.1.2, Nibabies >= 24.0.0a1.
            self._results["subject_sphere"] = os.path.join(
                self.inputs.segmentation_dir,
                participant_id,
                "freesurfer",
                participant_id,
                "surf",
                f"{hstr}.sphere.reg2",
            )
            # MCRIBS: tpl-fsaverage_hemi-?_den-41k_desc-reg_sphere.surf.gii
            self._results["source_sphere"] = os.path.join(
                self.inputs.segmentation_dir,
                "templates_fsLR",
                f"tpl-fsaverage_hemi-{hemisphere}_den-41k_desc-reg_sphere.surf.gii",
            )
            # MCRIBS: tpl-dHCP_space-fsaverage_hemi-?_den-41k_desc-reg_sphere.surf.gii
            self._results["sphere_to_sphere"] = os.path.join(
                self.inputs.segmentation_dir,
                "templates_fsLR",
                f"tpl-dHCP_space-fsaverage_hemi-{hemisphere}_den-41k_desc-reg_sphere.surf.gii",
            )
            # MCRIBS: tpl-dHCP_space-fsLR_hemi-?_den-32k_desc-week42_sphere.surf.gii
            self._results["target_sphere"] = os.path.join(
                self.inputs.segmentation_dir,
                "templates_fsLR",
                f"tpl-dHCP_space-fsLR_hemi-{hemisphere}_den-32k_desc-week42_sphere.surf.gii",
            )

        return runtime
```

File: xcp_d/interfaces/bids.py [hunk 1]
```diff
@@ -33,3 +42,144 @@ class DerivativesDataSink(BaseDerivativesDataSink):
...
+            # TODO: Collect from the preprocessing derivatives if they're a compliant version.
+            # Namely, fMRIPrep >= 23.1.2, Nibabies >= 24.0.0a1.
+            self._results["subject_sphere"] = os.path.join(
+                self.inputs.segmentation_dir,
+                participant_id,
+                "freesurfer",
+                participant_id,
+                "surf",
+                f"{hstr}.sphere.reg2",
+            )
```
Comment (tsalo) [inline xcp_d/interfaces/bids.py:165]: @madisoth so it ends up being in `/path/to/fmriprep/sourcedata/mcribs/sub-<id>/freesurfer/sub-<id>/surf/<hemi>.sphere.reg2`?

Comment (tsalo) [reply xcp_d/interfaces/bids.py:165]: I just want to make sure the extra `"freesurfer", participant_id` isn't a mistake.

Comment (madisoth) [reply xcp_d/interfaces/bids.py:166]: @tsalo when MCRIBS recon is used, there are actually two copies of the FreeSurfer-formatted output dir made:

1) the original:  `/sourcedata/mcribs/sub-<id>/freesurfer/sub-<id>`
2) the copy  `/sourcedata/freesurfer/sub-<id>` (i.e. same path as if FreeSurfer or infantFS recon was run)

Comment (tsalo) [reply xcp_d/interfaces/bids.py:166]: Ah good to know, thanks! We're grabbing the original here then? I'll move forward with your change then.

**Review by madisoth — APPROVED** [review]

**Pull request merged**